src/charm.py

```diff
@@ -602,6 +640,10 @@ def _on_config_changed(self, _: ConfigChangedEvent):
         # reconsider all data sent over the relations and all configs
         new_external_host = self.external_host
         new_routing_mode = self.config["routing_mode"]
+
+        if self.config["enable_experimental_forward_auth"]:
+            self.forward_auth.update_requirer_relation_data(self._forward_auth_config)
```

Comment:
Is this needed? Changes in the URL should not affect the `forward_auth` relation, also `_process_status_and_configurations` will be called if something changed.

Reply:
Right, I removed that part and added a check to call `_process_status_and_configurations` if the forward-auth config flag has changed.